conf/machine/include/imx-base.inc

```diff
@@ -653,11 +653,21 @@ WKS_FILE_DEPENDS:append:imx-generic-bsp:mx9-generic-bsp = " \
     ${@oe.utils.ifelse(d.getVar('UBOOT_PROVIDES_BOOT_CONTAINER') == '0', 'imx-boot', '')} \
 "
 
-SOC_DEFAULT_WKS_FILE ?= "imx-uboot-bootpart.wks.in"
-SOC_DEFAULT_WKS_FILE:mx8-generic-bsp ?= "imx-imx-boot-bootpart.wks.in"
+SOC_DEFAULT_WKS_FILE                 ?= "imx-uboot-bootpart.wks.in"
 SOC_DEFAULT_WKS_FILE:mxs-generic-bsp ?= "imx-uboot-mxs-bootpart.wks.in"
+SOC_DEFAULT_WKS_FILE:mx8-generic-bsp ?= "imx-imx-boot-bootpart.wks.in"
 SOC_DEFAULT_WKS_FILE:mx9-generic-bsp ?= "imx-imx-boot-bootpart.wks.in"
 
+IMX_BOOTPART_FILE_WKS                 = "${IMX_BOOTPART_FILE}"
+IMX_BOOTPART_FILE_WKS:imx-generic-bsp = \
+    "${@bb.utils.contains('PREFERRED_PROVIDER_u-boot', 'u-boot-imx', '${IMX_BOOTPART_FILE_TAGGED}', \
```

Comment:
I believe this is not going to work for vendor, bootloaders and others. I am wondering if we couldn't standardize the file name and use it for all.

Reply:
So instead of `u-boot-tagged.imx` and `imx-boot-tagged` you want a common filename, like `imx-bootpart-wic.bin`?

Reply:
I'm wondering if this is dependent of we using the downstream bootloader or the upstream bootloader. If this is the case, I believe that we should use a machine variable to decide if we are using one or another.

My understanding is that in the case with the upstream bootloader this is not necessary and work out of box. If this is the case, then we should mark this as U-Boot i.MX specific option and make that conditional to the value of a specific variable.

In the case of downstream, we can then set that accordingly if we are using the downstream or upstream versions. In the case of external vendors, they can set that in the machine accordingly.

How does that sound?